src/third-party/pinyin/pinyin_data.hpp

```diff
@@ -0,0 +1,454 @@
+    {U"mian", U"㒙㛯㝃㝰㡈㤁㨺㮌㰃㴐㻰䀎䃇䏃䛉䤄䩄䫵䯶䰓丏俛偭免冕冥勉勔喕娩婂媔嬵宀愐杣棉檰櫋汅沔泯渑湎湣澠牑眄眠瞑矈矊矏糆絻綿緜緡緬绵缅腼臱芇莬葂蝒蠠面靣靦鮸麪麫麵麺黽黾𡒳𡕢𡧍𡧒𡯫𢃮𢣔𣅍𣡠𣧾𥄝𥊿𥌂𥤵𥻩𥾝𦬛𦽃𧭇𧸨𨉥𨟺𨡞𨲛𨲾𩈹𩋠𩬞𩮉𩾃𪁼𪏊"},
+    {U"miao", U"㑤㠺㦝䁧䅺䏚䖢仯吵喵妙媌嫹庙庿廟彯描杪淼渺猫玅眇瞄秒竗篎紗緢緲繆缈苗藐蜱訬邈鱙鶓鹋𠋝𡙎𡡺𢆷𢚋𢤧𢷕𤾛𥭝𦳥𩳸𪃐𪃦"},
+    {U"mie", U"㒝㩢䁾䈼䌩䘊䩏乜吀咩咪哶孭幭懱搣櫗滅瀎灭烕眜篾羋蔑薎蠛衊覕鑖鱴鴓𠺗𡖺𡞙𡟬𢦼𢧞𢨖𤊾𤏿𥄲𥉓𥋚𥌨𥣫𥵒𥸴𥽘𥾝𦇪𧀅𧂝𨣱𩔠𩱷𪇴𪌺𪒍"},
+    {U"min", U"㞴㞶㟩㟭㢯㥸㨉㬆㳷䁕䂥䃉䋋䝧䟨䡑䡻䪸䲄僶冺刡勄厸呡垊姄岷崏忞怋悯惽愍慜憫抿捪敃敏敯旻旼暋民汶泯渂湏湣潣玟珉琘琝瑉痻皿盷盿眠砇碈笢笽簢緍緡繩缗罠苠蠠鈱錉鍲閔閩闵闽鰵鳘鴖黽𠊟𢼖𢽹𣱈𣱉𣷠𣹒𤇜𤛎𤸅𤺖𤿕𥜐𦈏𦌡𦫮𦳜𧁋𧌙𧲃𧵣𨏵𩔉𩭷𪂆𪄴𪉎"},
+    {U"ming", U"㝠㟰㫥䄙䆩䊅䏃䒌䫤䳟佲冥凕名命姳嫇慏掵明暝朙椧榠洺溟猽皿眀眳瞑茗萌蓂螟覭詺鄍酩銘铭鳴鸣𠋶𡥸𥌏𥥊𥹆𥿨𦡉𦫭𧟠𧱴𩣶𪂤𪗸"},
+    {U"miu", U"嘐繆謬谬𨱯"},
+    {U"mo", U"㱄㱳㵹㶬㷬㷵㹮䁼䁿䃺䊳䏞䒬䘃䜆䩋䬴䭩䮬䯢䱅䳮䴲万么伯佰冒劘劰勿唜嗼嘿嚤嚩嚰圽塻墨妺嫫嫼寞尛帓帕帞庅怽懡戂抹摩摸摹撫擵攠昧昩暯末枺模橅歾歿殁没沫湐漠瀎無爅狢獏瘼百皌眜眽眿瞐瞙砞磨礳秣粖糢絈絔縸纆耱膜艒茉莈莫蓦藐藦蘑蛨蟆蟔袜袹謨謩譕谟貃貈貉貊貌貘鄚銆鏌镆陌靺鞨饃饝馍驀髍魔魩魹麽麿默黙𠄮𠆮𠇱𠡞𠢓𠬛𠻚𡈗𡊉𡠜𡡉𡢜𡻟𡾉𢄏𢊗𢐖𢗿𢣗𢳀𣋟𣧒𣧣𣶊𣻕𤋂𤣻𤹴𤿖𥂓𥄕𥕓𥙎𥞪𥬎𥱹𥽘𥿉𦅔𦔭𦟟𦥦𦫕𦮅𧀅𧕤𧕥𧠓𧥟𧰱𧻙𧼟𧿴𨆽𨟖𨰞𨱱𩃁𩄻𩌧𩎟𩐻𩑦𩑷𩞁𩟠𩢖𩢷𩥔𩪮𩿣𪍇𪍤𪎠𪏟𪒂𪒇𬙊"},
+    {U"mou", U"㭌䋷䍒䍙䏬䗋䜼䥐䱕件侔劺厶呣哞堥婺恈敄某桙毋洠牟畝眸瞴繆缪蛑蟱袤謀谋踎鉾鍪鞪鴾麰𠀱𠥨𢃱𢜯𣫬𥆆𥿵𦊋𦊎𦋡𦭷𦳑𦺒𧎄𧎻𨴍𩢫𩶢"},
+    {U"mu", U"㙁㜈㟂㣎㧅㾇䀲䊾䑵䥈䱯亩仫凩募嘿坶墓墲姆姥娒婺峔幕幙慔慕拇暮木朰朷楘模樢母毣毪氁沐炑牟牡牧牳狇獏畆畒畝畞畮目睦砪穆縸繆胟艒苜茻莫莯萺蚞踇鉧鉬钼雮霂鞪鶩𠺖𡵬𢘃𢜯𢟨𣈊𣳠𤚅𤝂𤝕𤵝𥄈𥣸𥰻𦃤𦱒𧚀𧩒𧬏𧰷𧿹𨈶𨍎𨎸𨡭𨢢𩡨𩬍𩵦𩶖𩶩𪎫𬭁𬰃"},
+    {U"m̀", U"呣"},
+    {U"n", U"㕶咹哏哽唵嗯𠮾𧗈"},
+    {U"na", U"㗙㨥㪎㮏㵊䇱䈫䋈䋾䎎䏧䖓䖧䛔䟜䪏䫱乸内南吶呐呶哪嗱妠娜抐拏拿挐捺淰秅笚笝箬納絮纳肭蒘蒳衲袦訤詉誽豽貀蹃軜那郍鈉鎿钠镎雫靹魶𠕄𠱲𠴾𡤙𡰀𡷝𢇵𢓇𢜲𢡏𣅚𣡰𣸏𣹵𤓷𤔀𤝒𤡤𤬷𤭠𤱅𤱆𤷈𤸏𤸻𥍲𥑒𥹉𥿃𦙜𦛐𦣀𦬻𦰡𦳐𧋡𧘽𧤣𧦮𧰹𨙻𨚗𩏼𩚛𩟿𩮅𩮸𩹾𪌅𪐀𪗝"},
+    {U"nai", U"㜨㮈㮏㲡㴎㾍䍲䘅䯮乃佴倷哪奈奶妳嬭孻廼掜搱摨柰氖渿熋疓耏耐能腉艿萘螚褦迺釢錼鼐𠧟𠧤𡞫𡨵𡮙𢉓𣉘𣮦𥄗𥉃𦓎𦔹𦠸𦳐𦶅𨎡𩹟𪌞"},
+    {U"nan", U"㓓㫱㬮㽖䈒䊖䔜䛁䣸䩅䶲侽冉南喃囝囡妠娚婻嫨弇戁抩揇攤暔枏柟楠湳灘煵男畘罱腩莮萳蝻諵赧遖难難𡆤𡆱𡆲𢪈𢬷𤌔𤱣𤽲𤿏𦍀𦛚𦝧𦶈𧇙𧕴𧹞𨠹𨦳𨴌𨴘𨵴𩅠𩈑𩈶𩹞"},
+    {U"nang", U"㚂㶞䁸乪儾噥嚢囊囔憹搑擃攮曩欜涳瀼灢蘘蠰譨饢馕鬞齉𠶬𡿝𢖧𣰶𦈃𦗳𦣘𧅺𧖒𧟘𨳆𩛡𩜒"},
+    {U"nao", U"㑎㛴㞪㺀㺁㺒䃩䐉䛝䜀䜧䴃匘呶垴堖夒婥嫐孬峱嶩巎巙怓恼悩惱憹挠摎撓橈淖澆猱獶獿瑙硇碙碯脑脳腝腦膠臑蛲蝚蟯詉譊鐃铙閙闹鬧𠊦𠡷𡍍𡽧𡾂𡿺𢅈𢉵𢙐𢜸𢪼𣧽𣭺𣳦𤊲𤋫𤞍𤡤𤫕𤷻𥀮𥆲𥐻𥑪𥒢𦗮𧩣𧳀𧳦𧴓𧴙𨥸𨱵𩋈𩖯𩛋𩤘𩩀𩫔𩫺𩬷𩯆𫍢𬆛"},
+    {U"ne", U"㕯䅞䎪䭆呐呢哪抐疒疔眲訥讷那𢗉𣧍𧤜"},
+    {U"nei", U"㐻㨅㼏䲎內内娞婑氝浽脮腇那錗餒餧馁鮾鯘𠑚𠑛𡣢𢁩𢅼𢛉𣓃𥡭𨡌𩗔𩛞𩬀"},
+    {U"nen", U"㜛㯎㶧媆嫩嫰恁枘腝臑𡞾𧮠𨈗"},
+    {U"neng", U"㲌㴰䏻竜而耐能螚𠹌𢆂𨃳𨶙"},
+    {U"ng", U"㕶哽唔唵嗯"},
+    {U"ni", U"㘈㞾㠜㥾㦐㩘㪒㮏㲻㵫㹸䁥䕥䘌䘦䘽䛏䝚䦵䵑䵒伱伲你倪儗儞兒匿呢坭埿堄妮妳婗嫟嬭嬺孨孴尼屔屰嶷彌怩惄愵慝懝抐抳拟掜擬旎昵晲暱柅棿檷氼泥淣溺濔濘瀰灄爾狔猊痆眤睨禰秜籾縌聣聻胒腝腻膩臡苨薿蚭蛪蜺觬誽譺貎跜輗迡逆郳鈮鉨鑈铌隬霓馜鯢鲵麑齯鿭𠆵𠒰𠱘𠸺𠽬𡎳𡎿𡞭𡣁𡥦𡥨𡫸𡬗𢅟𢘝𢚮𢛜𢣚𢦱𣘗𣡋𣢞𣭙𣲷𤙌𤦤𥄽𥇄𥜦𥜬𥷄𥺜𦆦𦤽𦦃𦮾𦰫𧃩𧈞𧏾𧖷𧵼𧺰𨀀𨋗𨒫𨺙𨽦𩈢𩉹𩋪𩍦𩚮𩚯𩩢𩯨𩰞𩱄𩸦𩸧𩺝𩺱𩾆𪏵𪏸𪐌𪙛𫐐𫠜"},
+    {U"nia", U"㖸"},
+    {U"nian", U"㜤㞋㮟㲽䄭䄹䚓䧔䬯䴴卄哖唸埝姩年廿念拈捵捻撚撵攆榐涊淰溓焾痆碾秊秥簐粘艌蔫趁跈蹍蹨躎輦輾辇辗鮎鯰鲇鲶鵇黏𠕟𠗋𠣇𠫺𠬋𡝟𡰫𣎔𣐏𤁥𤽿𥮘𥺴𦁇𦭁𦷙𧹞𨀀𨇍𨋚𨏛𨚶𨢯𨴞𨸱𩉄𩊫𩽴𪐇𪑮"},
+    {U"niang", U"䖆娘嬢孃酿醸釀𥽬𪓃"},
+    {U"niao", U"㒟㜵㞙㠡㭤㳮䃵䙚䦊䮍嫋嬝嬲尥尿樢溺脲茑茮蔦袅裊褭鳥鸟𠒰𡘏𡝋𡝒𡠿𢶑𢸣𣟊𤲶𥤂𥾇𨳀𨽖𩖔𩭑𪅝𪈼"},
+    {U"nie", U"㖏㖕㖖㘝㘨㘿㙞㚔㜸㡪㩶㮆㴪㸎䂼䄒䇣䌜䌰䜆䡾䭃䯀䯅䯵䳖乜倪哪啮喦嗫噛嚙囁囐囓囡圼埝孼孽峊嵒嵲嶭巕帇幸惗捏捻掜揑摰攝敜枿棿槷櫱泥涅湼痆篞籋糱糵聂聶臬臲苶菍蘖蠥褹諗讘踂踗踙蹑躡鉨鉩銸鋷錜鎳鑈鑷钀镊镍闑陧隉顳颞齧𠈊𠶿𡆣𡇄𡍤𡰆𡴎𡶫𡸣𡾦𡾲𡿖𡿗𢈸𢫻𢬴𣀳𣌍𣙗𣯭𣰼𤭂𤴘𤶚𤷈𤸏𤺐𥔄𥬞𥬬𥭙𥮤𦄌𦈙𦘒𦛠𦞆𦡃𦣀𦯖𦵐𧁈𧋖𧞍𧻼𨊞𨙓𨱺𨲀𨶠𨻄𩋏𩐭𩒕𩖁𩣘𪌊𪌿𪎃𪎅𪘎𫔶𬛸"},
+    {U"nin", U"㤛䋻䚾䛘囜恁您拰脌𠽝"},
+    {U"ning", U"㝕㣷㲰㿦䆨䔭䗿䭢佞侫倿儜冰凝咛嚀嬣宁寍寕寗寜寧年拧擰攘柠橣檸泞泥澝濘狞獰甯疑矃聍聹苧薴鑏鬡鬤鸋𡫃𡷪𣍆𤕦𤹧𤻝𥣗𥧤𥳥𦡲𦡼𧃱𧍖𧑗𧕝𧭈𩕳𫛢"},
+    {U"niu", U"㖻㺲㽱䂇䋴䏔䒜妞忸怓扭拗杻汼沑炄牛牜狃紐纽莥蚴鈕钮靵𢔟𣅚𣧊𣲶𥀝𥍳𥝦𧘥𨋀𨙺𨳞𨷁𩈇𩙷𩚖𩜷𩲍𩵠𪏲"},
+    {U"nong", U"㶶㺜䁸䢉䵜侬儂农咔哝噥弄憹挊挵檂欁浓濃燶癑禯秾穠繷脓膿莀蕽襛農辳醲齈𠌚𠘊𥂒𨑊𨠤𨲳𩅽𩇔𩟊𪆯𪒬𬪩"},
+    {U"nou", U"㜌㝹㳶䅶䘫䨲䰭啂嬬搙擩槈檽獳羺耨譨譳鎒鐞𠲴𡝦𡨻𡭾𢉕𢉚𣻖𤟦𥀫𧂦𧃨𧅘𩆟𩒔𪋺"},
+    {U"nu", U"㚢㣽仅伮傉努呶奴孥帑弩怒挐搙擩砮笯肭胬褥詉駑驽𠴂𢪦𢫓𢫭𥅄𥛑𥤨𥱂𧉭𧗈𧪅𧿔𪺹"},
+    {U"nuan", U"㬉䙇奻暖渜湪濡煖煗臑餪𪋐"},
+    {U"nun", U"媆黁"},
+    {U"nuo", U"㐡㑚㔮㖠㛂㡅㰙䇔䎠䚥傩儺呐哪喏堧娜媠愞懦懧挪掉掿搙搦搻梛榒橠毭濡稬穤糑糥糯耎袲袳諾诺蹃逽那郍鍩锘難需𠸱𠹈𡖔𡖫𡬥𡿊𢜪𢰜𢾲𣃽𣆚𤘟𥑽𥻾𦀨𦂍𦓢𦡃𦩜𧣚𧣺𨁌𨎭𩈺𩴓𩷁"},
+    {U"nv", U"㵖䏔䖡䘐䚼䶊女恧朒沑狃籹絮聏胬衂衄釹钕𥄋𥍞𦓕𦓖𧖷𩈢"},
+    {U"nve", U"䖈䖋䨋婩疟瘧硸虐𤷡𨵫"},
```

Comment:
I think "nue" is more common when typing with an IME? There are also other possible alternative spellings such as "lve" - "lue", "jv" - "ju" so we should probably create an alternative spelling map and use it when building the character-pinyin map in pinyin.cpp.

Reply:
Mapped ü to both u and v, hopefully that will make more sense to end user